### PR TITLE
Fix RoleBindings form & EventsList

### LINKED ISSUE #1846: Events at CRD List level are not contextual
**Description**

The events provided at the bottom of the CRD should be contextual

**Expected result**

The events provided at the bottom of the CRD should be contextual

**Actual result**

The events provided at the bottom of the CRD are general.

**Steps to reproduce**

**Troubleshooting**

### LINKED ISSUE #1848: Wrong placeholder text for Create RoleBinding
**Description**

**Expected result**

Wrong placeholder text for Create RoleBinding
**Actual result**

Should say Select Roles from list

**Steps to reproduce**

**Troubleshooting**

image

### LINKED ISSUE #1849: Role Selection from RoleBinding does not allow free text
**Description**

The Form Validation fails if free text is typed in

**Expected result**

User should be able to add free text and the form should be valid

**Actual result**

**Steps to reproduce**

**Troubleshooting**

image

### PR DESCRIPTION
**Pull request by Wawrzyn321**
**Description**

Changes proposed in this pull request:

- Fix small bug in Role template (`Role` cannot have `nonResourceUrls`)
- #1849 - separate `Roles` and `ClusterRoles` combobox into a type dropdown (R or CR) and name combobox.
- Combine filters (type & the one passed upstream) in EventsList (#1846)

**Related issue(s)**

Closes #1849
Closes #1848
Closes #1846

### CODE REVIEW

**Comment by Lyczeq** [comment]
There is a bug when changing the `Role Type`. For example, I set `Role` for `Role type` and then select `serverless-build` as a `Role`. When I switch from `Role` to `ClusterRole`, the `serverless-build` stays the same which is wrong.

File: core-ui/src/resources/Events/EventList.js [hunk 1]
```diff
@@ -120,8 +120,14 @@ export function EventList(props) {
...
       readOnly={true}
       filter={res => {
-        if (displayType.key === EVENT_MESSAGE_TYPE.ALL.key) return true;
-        return res.type === displayType.key;
+        const typeFilter =
+          displayType.key === EVENT_MESSAGE_TYPE.ALL.key ||
+          res.type === displayType.key;
+
+        const propsFilter =
+          typeof props.filter === 'function' ? props.filter(res) : true;
```
Comment (Lyczeq) [inline core-ui/src/resources/Events/EventList.js:128]: Just an idea, have you thought about extracting `filter` and `type` from the props? The `props` prefix won't be needed then.

**Comment by Lyczeq** [comment]
I've just encountered one thing which is related to the Roles form. The `Resources` section has a button `Add all` but it does nothing when clicked at the beginning. Maybe we should consider disabling it? 
image

**Review by Lyczeq — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**